packages/@n8n/ai-workflow-builder.ee/src/ai-workflow-builder-agent.service.ts

```diff
@@ -190,10 +200,69 @@ export class AiWorkflowBuilderService {
 
 	async *chat(payload: ChatPayload, user: IUser, abortSignal?: AbortSignal) {
 		const agent = await this.getAgent(user);
+		const userId = user?.id?.toString();
+		const workflowId = payload.workflowContext?.currentWorkflow?.id;
 
-		for await (const output of agent.chat(payload, user?.id?.toString(), abortSignal)) {
+		for await (const output of agent.chat(payload, userId, abortSignal)) {
 			yield output;
 		}
+
+		// After the stream completes, track telemetry
+		if (this.onTelemetryEvent && userId) {
+			try {
+				await this.trackBuilderReplyTelemetry(agent, workflowId, userId);
+			} catch (error) {
+				this.logger?.error('Failed to track builder reply telemetry', { error });
+			}
+		}
+	}
+
+	private async trackBuilderReplyTelemetry(
+		agent: WorkflowBuilderAgent,
+		workflowId: string | undefined,
+		userId: string,
+	): Promise<void> {
+		if (!this.onTelemetryEvent) return;
+
+		// Get the final state from the agent
+		const state = await agent.getState(workflowId, userId);
+		const threadId = SessionManagerService.generateThreadId(workflowId, userId);
+
+		// Extract the last AI message using instanceof
+		const aiMessages = state.values.messages.filter(
+			(m: BaseMessage): m is AIMessage => m instanceof AIMessage,
+		);
+		const lastAiMessage = aiMessages[aiMessages.length - 1];
```

Comment:
Could be a single step
```suggestion
		const lastAiMessage = state.values.messages.findLast(
			(m: BaseMessage): m is AIMessage => m instanceof AIMessage,
		);
```